src/items/__tests__/item.service.spec.ts

```diff
@@ -256,5 +256,37 @@ describe('ItemService', () => {
         expect(actual).toEqual(expected);
       });
     });
+    describe('and the execution is unsuccessful', () => {
+      it('throw a NotFoundException when item is not found', async () => {
+        const unexistingItemId = '123';
+        const unexistingAddress = 'address';
+        const errorMessage = `The item with id ${unexistingItemId} does not exist`;
+
+        itemRepository.findById.mockResolvedValue(null);
+
+        const exception = () =>
+          service.delistAnItem({ itemId: unexistingItemId, address: unexistingAddress });
+
+        await expect(exception).rejects.toThrow(NotFoundException);
+        await expect(exception).rejects.toEqual(new NotFoundException(errorMessage));
+
+        expect(itemRepository.delistAnItem).not.toHaveBeenCalled();
+      });
+      it('throw a NotFoundException when account is not found', async () => {
```

Comment:
You have two its inside the same describe scenario. Normally you have one it per scenario